engine.py

```diff
@@ -198,6 +209,113 @@ def log_warning(self, msg):
+            return panel_info
+
+        return None
+
+    def show_panel(self, panel_id, title, bundle, widget_class, *args,
+        **kwargs):
+        """Show the panel matching the supplied args. 
+
+        Will first try to locate an existing instance of the panel. If it 
+        exists, it will make it current. If it can't find an existing panel,
+        it will create a new one.
+
+        If the panel can't be created for some reason, the widget will be 
+        displayed as a dialog.
+        
+        :param panel_id: Unique id to associate with the panel - normally this
+            is a string obtained via the register_panel() call.
+        :param title: The title of the window
+        :param bundle: The app, engine or framework object that is associated
+            with this window 
+        :param widget_class: The class of the UI to be
+        constructed. This must derive from QWidget.
+        
+        Additional parameters specified will be passed through to the
+        widget_class constructor.
+        """
+
+        # check to see if we just need to return the widget itself. Since we
+        # don't really have information about the panel outside of this call,
+        # we use a special flag to know when the info is needed and return it.
+        if hasattr(self, '_panel_info_request') and self._panel_info_request:
+            return {
+                'id': panel_id,
+                'title': title,
+                'bundle': bundle,
+                'widget_class': widget_class,
+                'args': args,
+                'kwargs': kwargs,
+            }
+
+        # try to locate the pane in the desktop and make it the current tab. 
+        for pane_tab in hou.ui.curDesktop().paneTabs():
+            if pane_tab.name() == panel_id:
+                pane_tab.setIsCurrentTab()
+                return
+
+        # panel support differs between 14/15. 
+        if self._panels_supported():
+
+            # if it can't be located, try to create a new tab and set the
+            # interface.
+            panel_interface = None
+            try:
+                for interface in hou.pypanel.interfacesInFile(self._panels_file):
+                    if interface.name() == panel_id:
+                        panel_interface = interface
+                        break
+            except hou.OperationFailed:
+                # likely due to panels file not being a valid file, missing, etc. 
+                # hopefully not the case, but try to continue gracefully.
+                hou.log_warning(
+                    "Unable to find interface for panel '%s' in file: %s" % 
+                    (panel_id, self._panels_file))
+
+            if panel_interface:
+                # the options to create a named panel on the far right of the
+                # UI doesn't seem to be present in python. so hscript it is!
+                hou.hscript("pane -S -m pythonpanel -o -n %s" % panel_id)
```

Comment:
You could totally argue that it's overkill, but i personally find it super useful to add a comment with a link to the docs when executing commands with parameters like this. Makes it easier to do future maintenance.